Modules/symtablemodule.c

```diff
@@ -71,13 +71,57 @@ static PyMethodDef symtable_methods[] = {
     {NULL,              NULL}           /* sentinel */
 };
 
+static int
+symtable_init_stentry_type(PyObject *m)
+{
+    return PyType_Ready(&PySTEntry_Type);
+}
+
+static int
+symtable_init_constants(PyObject *m)
+{
```

Comment:
I'm trying to slowly use macros to factorize the code, something like:

```
#define ADD_INT(value) \
    if (PyModule_AddIntConstant(module, #value, value) < 0) { \
        return -1; \
    }
```

A more generic machinery like the one you proposed would avoid such macro, and until there is such API, IMO using a macro makes the code more readable:

```
ADD_INT(USE);
```

You can add a second macro for PyModule_AddIntConstant() calls.

Reply:
I'd rather not spend more time then necessary. This was a simple search and replace job.

Let's consider refactoring after ``PyModuleConstants_Def`` from https://github.com/python/cpython/pull/23286 has landed. The new feature will make module constant definitions easier.